packages/main/src/Button.ts

```diff
@@ -382,6 +389,12 @@ class Button extends UI5Element implements IFormElement {
 		this.buttonTitle = this.tooltip || await getIconAccessibleName(this.icon);
 	}
 
+	onAfterRendering() {
+		if (this._keepActiveState) {
```

Comment:
Setting a property from the SPlitButton to the Button is tight coupling. Better fire an private event in the Button "active-change" that the SplitButton listens to and can react/prevent so that the correct (effective) active state is properly maintained.